pkg/policy/repository.go

```diff
@@ -139,6 +141,69 @@ type Repository struct {
 	getEnvoyHTTPRules func(certificatemanager.SecretManager, *api.L7Rules, string) (*cilium.HttpNetworkPolicyRules, bool)
 }
 
+// getShardKey returns the key to be used for the sharding of Repository.rulesByShard.
+// We currently use the K8s namespace label as the key, and if it's not present, we fall back
+// to an empty string. Returns a true if it found a namespace label, and false if not.
+func getShardKey(lbls labels.LabelArray) (string, bool) {
+	key := ""
+	matched := false
+	for _, lbl := range lbls {
+		if lbl.Key == k8sConst.PolicyLabelNamespace && lbl.Source == labels.LabelSourceK8s {
+			// If we find two, ignore then and put them into the non-namespaced for easier housekeeping
+			if matched {
+				return "", false
+			}
+			key = lbl.Value
+			matched = true
+		}
+	}
+	return key, matched
```

Comment:
The logic in `getNamespaceFromLabels` and `getShardKey` seems duplicative, might be worth implementing one in terms of the other.


```suggestion
	if key := getNamespaceFromLabels(lbls); key != "" {
		return key, true
	}
	return "", false
```

Reply:
Sadly those two functions use different labels; since there is one label that is present in endpoint selectors (`io.kubernetes.pod.namespace`), while there is another one in the `rule.Labels` field (`io.cilium.k8s.policy.namespace`).

Reply:
I was thinking we could reuse the k8s label search logic by adjusting the function definition for reuse. The refactor I had in mind is a bit more than fits in github suggestion, so I created https://github.com/odinuge/cilium/pull/94